microsoft/testsuites/dpdk/dpdkovs.py

```diff
@@ -111,42 +111,49 @@ def _check_ovs_dpdk_compatibility(self, dpdk_tool: DpdkTestpmd) -> None:
         # confirm supported ovs:dpdk version pairing based on
         # https://docs.openvswitch.org/en/latest/faq/releases/
         # to account for minor releases check release is below a major version threshold
-        ovs_dpdk_version_pairings = [
-            ("2.13.0", 18),  # if below 2.13.0 then dpdk version is 18
-            ("2.15.0", 19),  # and so on
-            ("2.17.0", 20),
-            ("2.18.0", 21),
-        ]
+        dpdk_to_ovs_minimum_version = {
+            19: "2.13.0",
+            20: "2.15.0",
+            21: "2.17.0",
+            22: "3.0.0",
+        }
         # check is version too low
         if self.ovs_version < "2.11.0":
             fail(
                 "OVS < 2.10 is not supported with any supported "
                 "DPDK version as of 2/8/2022"
             )
-        # check if version too high (pairing is unknown)
-        elif self.ovs_version > "2.18.0":
-            raise NotImplementedError(
-                (
-                    f"Not implemented OVS version {self.ovs_version}. "
-                    "Please update the version requirement table in "
-                    "dpdkovs.py:check_ovs_dpdk_compatibilty with new data"
-                    "from https://docs.openvswitch.org/en/latest/faq/releases/"
+        else:
+            if dpdk_version < "19.11.0":
+                raise SkippedException(
+                    f"Dpdk version {dpdk_version} is not supported by this test."
+                )
+
+            if int(dpdk_version.major) not in dpdk_to_ovs_minimum_version.keys():
+                # we've already checked out latest OVS
+                # DPDK version is above 22, warn and proceed.
+                self.node.log.warn(
+                    "DPDK version is above the maximum in the version match"
+                    " table. Using latest OVS."
                 )
+                return
```

Comment:
Does it mean not to run following ovs enlistment code? So what happens, if return here?

Reply:
OVS is already checking out latest at the top of the test, so if the DPDK version is also above the levels in our table (22 and beyond) it should be fine to just run with what's already checked out. I'll comment this a little better.
I don't really like that this table has to be updated constantly, but if it starts failing we'll see the note in the log and know that the test might need an update to the table.

Reply:
The alternative is pulling the rst table from the docs and parsing it though, so this seems cleaner than that.

Reply:
Fantastic version compatibility handling!